### PR TITLE
Fix user preference persistence for Gradio UI

### PR DESCRIPTION
**Pull request by SeriousBuggie**
Fix save and restore of user preferences via localStorage. Previously, dropdown values were saved as display labels instead of choice values, dropdown changes via Gradio's programmatic value setter were not detected, and restored preferences were silently discarded on page reload because Gradio does not forward JS return values to Python when `inputs=None`.

This PR fixes all three issues: dropdown save now maps labels to values using embedded choice data, a value property wrapper catches programmatic updates, and restore uses a hidden dummy Textbox as a data channel. MP3 control visibility is moved to client-side DOM sync since Gradio does not fire `.change()` for load-time assignments.

## Summary by CodeRabbit

* **Refactor**
  * More deterministic preference restore and wiring; restored payloads are parsed as a single JSON payload and MP3 visibility is synced from the save-side.
  * Centralized dropdown metadata and consistent MP3 control visibility handling.

* **Bug Fixes**
  * Fixed syncing and show/hide behavior for MP3 bitrate/sample-rate controls and made dropdown mapping parsing more resilient.

* **Tests**
  * Expanded tests for dropdown mappings, restore behavior, invalid payloads, and edge cases.

### CODE REVIEW

File: acestep/ui/gradio/interfaces/user_preferences.js [hunk 1]
```diff
@@ -75,6 +75,19 @@
...
+                try {
+                    const mapsData = JSON.parse(maps.dataset.maps);
+                    const value = el.value;
+                    const pair = mapsData.find(item => item[0] === value);
+                    if (pair !== undefined) return pair[1];
+                } catch (e) {
+                    console.warn(`[UserPreferences] Failed to parse mapping for ${spec.elemId}:`, e);
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (SeriousBuggie) [reply acestep/ui/gradio/interfaces/user_preferences.js:90]: Wrong comment. el.value actually store a **label**, not a **key** for dropdowns Gradio. So no any mistake here.
